### ISSUE TITLE
Randomly failing test: CarrierReaderFromCSVTest.carrierCreation()

### ISSUE THREAD

**Issue by michalmac** [matsim-org/matsim-libs#1976]
I noticed this test fails from time to time:
- https://ci.appveyor.com/project/michalmac/matsim-libs/builds/43511360
- http://ci.matsim.org:8080/job/MATSim_GitHub/6502/

Sometimes the number of generated vehicles for the second carrier is 6 instead of 9:
```
java.lang.AssertionError: expected:<9> but was:<6>
	at org.matsim.freightDemandGeneration.CarrierReaderFromCSVTest.carrierCreation(CarrierReaderFromCSVTest.java:107)
```

Here is the random component that makes the vehicle creation non-deterministic/non-reproducible: https://github.com/matsim-org/matsim-libs/blob/30e66e61782444c9350ad43f7a7c93087f557203/contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java#L467-L478

Sometimes the randomly chosen depot links are identical (i.e. located at the same link). Because vehicle ids are created using depot ids, having identical depots leads to creating vehicles with duplicated ids, and because vehicles are stored in a `Map<Id<Vehicle>, CarrierVehicle>`, only one vehicle with the same id will be inserted.

https://github.com/matsim-org/matsim-libs/blob/30e66e61782444c9350ad43f7a7c93087f557203/contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java#L487-L497

Suggested fix: make sure depots are unique (e.g. use Set<String> or check uniqueness). Alternatively, generate vehicle ids so that they are unique.

Side note: Maybe consider failing if not all depots are provided in the CSV (instead of choosing them randomly).

**Comment by rewertvsp** [comment]
Thanks for noticing this duplicating problem. Now the vehicleIds should be unique.
https://github.com/matsim-org/matsim-libs/blob/8e235a70d69057a05549beab011b019bb6eabb62/contribs/application/src/main/java/org/matsim/freightDemandGeneration/CarrierReaderFromCSV.java#L479-L510
Regarding your note: I still have the random option, because if you want to generate small scale commercial traffic, you need hundreds of depots. Therefore, the random choice is helpful.
PR: #1978

**Comment by michalmac** [comment]
fixed via #1978
